test/database/skylink_test.go

```diff
@@ -0,0 +1,87 @@
+package database
+
+import (
+	"context"
+	"testing"
+
+	"github.com/skynetlabs/pinner/conf"
+	"github.com/skynetlabs/pinner/database"
+	"github.com/skynetlabs/pinner/test"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+// TestSkylink is a comprehensive test suite that covers the entire
+// functionality of the Skylink database type.
+func TestSkylink(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	if conf.ServerName == "" {
+		conf.ServerName = test.TestServerName
+	}
+
+	ctx := context.Background()
+	dbName := test.DBNameForTest(t.Name())
+	db, err := test.NewDatabase(ctx, dbName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sl := test.RandomSkylink()
+
+	// Fetch the skylink from the DB. Expect ErrSkylinkNoExist.
+	_, err = db.SkylinkFetch(ctx, sl)
+	if !errors.Contains(err, database.ErrSkylinkNoExist) {
+		t.Fatalf("Expected error %v, got %v.", database.ErrSkylinkNoExist, err)
+	}
+	// Try to create an invalid skylink.
+	_, err = db.SkylinkCreate(ctx, "this is not a valid skylink", conf.ServerName)
+	if err == nil {
+		t.Fatal("Managed to create an invalid skylink.")
+	}
+	// Create the skylink.
+	s, err := db.SkylinkCreate(ctx, sl, conf.ServerName)
+	if err != nil {
+		t.Fatal("Failed to create a skylink:", err)
+	}
+	// Validate that the underlying skylink is the same.
+	if s.Skylink != sl {
+		t.Fatalf("Expected skylink '%s', got '%s'", sl, s.Skylink)
+	}
+	// Add the skylink again, expect ErrSkylinkExists.
+	_, err = db.SkylinkCreate(ctx, sl, conf.ServerName)
+	if !errors.Contains(err, database.ErrSkylinkExists) {
+		t.Fatalf("Expected '%s', got '%v'", database.ErrSkylinkExists, err)
+	}
+	// TODO Add the same skylink again, this time in base32. Make sure it doesn't get duplicated.
```

Comment:
Did you mean to add this in this MR or f/u with it?

For reference we don't allow TODOs in skyd because we have come to the realisation that they clutter the codebase because we simply forget about them. 

This is the first MR for this repo so it's fine but in this case it's not clear whether you intended to add that or if it's just a leftover TODO from a different repo that we never implemented.

Reply:
The idea was to do it in a F/U. I actually ended up using skyd's `skylink.LoadString`, making this redundant.